File: QE-selenium/java/Activity8.java
package demos;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        Actions actions = new Actions(driver);

        driver.get("https://training-support.net/webelements/mouse-events");
        System.out.println(driver.getTitle());

        WebElement cargoLockButton = driver.findElement(By.xpath("//*[@id='stage']/div[1]/div[1]/div"));
        actions.click(cargoLockButton).perform();

        WebElement cargoTomlButton = driver.findElement(By.xpath("//h1[text()='Cargo.toml']/parent::div"));
        actions.click(cargoTomlButton).perform();

        System.out.println(driver.findElement(By.id("result")).getText());

        WebElement srcButton = driver.findElement(By.xpath("//h1[text()='src']/parent::div"));
        actions.doubleClick(srcButton).perform();

        WebElement targetButton = driver.findElement(By.xpath("//h1[text()='target']/parent::div"));
        actions.contextClick(targetButton).perform();

        WebElement openButton = driver.findElement(By.xpath("//*[@id='menu']/div/ul/li[1]"));
        actions.click(openButton).perform();

        System.out.println(driver.findElement(By.id("result")).getText());

        driver.quit();
    }
}

File: QE-selenium/python/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.action_chains import ActionChains

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/mouse-events")
    print(driver.title)
    cargo_lock_button = driver.find_element(By.XPATH,'//*[@id="stage"]/div[1]/div[1]/div')
    ActionChains(driver).click(cargo_lock_button).perform()
    cargo_to_button = driver.find_element(By.XPATH,'//h1[text()="Cargo.toml"]/parent::div')
    ActionChains(driver).click(cargo_to_button).perform()
    print(driver.find_element(By.ID,"result").text)
    src_button = driver.find_element(By.XPATH,'//h1[text()="src"]/parent::div')
    ActionChains(driver).double_click(src_button).perform()
    target_button = driver.find_element(By.XPATH,'//h1[text()="target"]/parent::div')
    ActionChains(driver).context_click(target_button).perform()
    open_button = driver.find_element(By.XPATH,'//*[@id="menu"]/div/ul/li[1]')
    ActionChains(driver).click(open_button).perform()
    print(driver.find_element(By.ID,"result").text)
    driver.quit()